pkg/controller/tas/topology_ungater.go

```diff
@@ -447,17 +446,17 @@ func readRanksForLabels(
 	podIndexLabel string,
 	rjInfo *replicatedJobsInfo,
 ) (map[int]*corev1.Pod, error) {
-	result := make(map[int]*corev1.Pod, 0)
+	result := make(map[int]*corev1.Pod)
 	podSetSize := int(*psa.Count)
 	for _, pod := range pods {
-		podIndex, err := readIntFromLabel(pod, podIndexLabel)
+		podIndex, err := utilpod.ReadUIntFromLabel(pod, podIndexLabel)
```

Comment:
Let's use `ReadUIntFromLabelBelowBound(pod, podIndexLabel, singleJobSize)` here. The bound value is `singleJobSize`. Let's compute on L451 
```golang
singleJobSize := podSetSize
if rjInfo != nil {
  singleJobSize := podSetSize / rjInfo.replicasCount
}
```
then drop L464.

Reply:
Done